.github/workflows/_reusable_run_tests.yml

```diff
@@ -114,12 +120,17 @@ jobs:
         run: |
           bundle exec fastlane prepare_for_tests
 
-      - name: "Test all selected frameworks" 
+      - name: "Test new critical flows"
+        run: |
+          bundle exec fastlane run_uitests
+
```

Comment:
as for a test I put it here, but as it would replace the critical flows, we could run it on each PR

like:
```
if: ${{ inputs.test_dependencies || inputs.all }}
```